persistence-zookeeper/src/test/scala/com/mesosphere/usi/repository/InMemoryRepositoryTest.scala

```diff
@@ -0,0 +1,13 @@
+package com.mesosphere.usi.repository
+
+import com.mesosphere.utils.UnitTest
+import com.typesafe.scalalogging.StrictLogging
+
+class InMemoryRepositoryTest extends UnitTest with RepositoryBehavior with StrictLogging {
```

Comment:
I'm confused by the location of this file (and other _in-memory-repository_ related files). Why are those in the `persistence-zookeeper` sub-module? Why is `InMemoryPodRecordRepository.scala` in the `persistence` sub-module? Ideally:
```
usi
  |
  ...
   - persistence # traits and interfaces, no implementation
   - persistence-zookeeper # zookeeper implementation
   - test-utils # In-memory implementation here (see DummyMetrics for example)? 
      It is not supposed to be used in production so this seems appropriate
```

Reply:
I guess it was done so that we don't need to have 
```java
implementation project(':test-utils')
```
and can just do 
```java
implementation project(':persistence')
testCompile project(':test-utils')
```
in order to instantiate the persistence layer. I have moved `InMemoryPodRecordRepository` to `test-utils` and added a note to `build.gradle` file to replace the `implementation project(':test-utils')` with `testCompile project(':test-utils')` when we have a `persistence-zookeeper` implementation. 

The only thing I am unsure about is the location of `InMemoryRepositoryTest` which is currently in `persistence-zookeeper`. I can not move this to `test-utils` as it would cause a cyclic dependency. I guess having this here is _fine_ as its intention is to verify the `RepositoryBehavior` which would be used to verify a zk persistence layer in future. wdyt ?